lib/gulp-image-converter.js

```diff
@@ -0,0 +1,64 @@
+var path = require('path');
+var _ = require('underscore');
+var through = require('through2');
+var lwip = require('lwip');
+var gutil = require('gulp-util');
+
+function convert(options, file, cb) {
+  var format = path.extname(file.path).slice(1);
+  var targetFormat = options.format || format;
+
+  lwip.open(file.contents, format, function(err, image) {
+    if (err) {
+      return cb(err);
+    }
+
+    image.toBuffer(
+      targetFormat,
+      options.formatParams || {},
+      function(err, buffer) {
+        if (err) {
+          return cb(err);
+        }
+
+        // Note that vinyl 0.5 supports file.extname for manipulations
+        // like this, but we're not yet using 0.5 at the time
+        // of writing this, so we'll need to replace the extension
+        // ourselves.
+        file.path = require('replace-ext')(file.path, '.' + targetFormat);
+        file.contents = buffer;
+
+        gutil.log(
+          'converted ' +
+          gutil.colors.cyan(path.basename(file.relative)) + ' (' +
+          (Math.floor(buffer.length / 1024)) + 'k)'
+        );
+        cb(null, file);
+      }
+    );
+  });
+}
+
+module.exports = function(options) {
+  var fileMap = options.files;
+  var convertedFiles = [];
+
+  return through.obj(function(file, enc, cb) {
+    var posixRelative = file.relative.replace(/\\/g, '/');
+    var convertOptions = fileMap[posixRelative];
+    if (!convertOptions) {
+      return cb(null, file);
+    }
+    if (file.isStream()) {
```

Comment:
Oh, yeah, basically gulp's vinyl file objects can have their contents be either buffers or streams... by default they're buffers, but sometimes, if you're dealing with super massive files that can't fit into memory, or if your plugin uses libraries that work with streams instead of buffers, it may be better to use streams instead. That can be done from the gulpfile by writing e.g. `gulp.src('*.js', { buffer: false })`.

In our case, we're using buffers because that's what `lwip` operates on, so it's just easiest to write a plugin that only works with buffers (especially since we don't currently have any plans to make this plugin reusable by other folks).

Anyways, since we don't pass `{ buffer: false }` into the stream that we give to this plugin in our gulpfile, the error will never get raised, but it's useful for documentation purposes to keep that assertion in there, like the [gulp docs on using buffers](https://github.com/gulpjs/gulp/blob/master/docs/writing-a-plugin/using-buffers.md) do.